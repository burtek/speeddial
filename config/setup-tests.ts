import '@testing-library/jest-dom';
import { createSerializer } from '@emotion/jest';
import i18n from 'i18next';
import 'jest-extended';
import 'jest-extended/all';
import { initReactI18next } from 'react-i18next';


expect.addSnapshotSerializer(createSerializer());

await i18n
    .use(initReactI18next)
    .init({
        lng: 'dev',
        interpolation: { escapeValue: false }
    });
