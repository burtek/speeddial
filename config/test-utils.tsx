import type { RenderOptions } from '@testing-library/react';
import { render } from '@testing-library/react';
import type { FC, PropsWithChildren, ReactElement } from 'react';
import { Provider as StoreProvider } from 'react-redux';

import { store } from '@@data/index';
import { ThemedApp } from 'src/theme/themed-app';

const AllTheProviders: FC<PropsWithChildren> = ({ children }) => (
    <StoreProvider store={store}>
        <ThemedApp>
            {children}
        </ThemedApp>
    </StoreProvider>
);

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
// eslint-disable-next-line import/export
export * from '@testing-library/react';

// override render method
// eslint-disable-next-line import/export
export { customRender as render };
