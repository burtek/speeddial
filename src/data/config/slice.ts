import type { PayloadAction } from '@reduxjs/toolkit';
import { createSlice } from '@reduxjs/toolkit';

export type ThemeMode = 'dark' | 'light';

export const initialState = () => ({
    modalOpen: false,
    themeMode: null as ThemeMode | null
});

export const { actions, name: sliceName, reducer } = createSlice({
    name: 'config',
    initialState,
    reducers: {
        openModal(state) {
            state.modalOpen = true;
        },
        closeModal(state) {
            state.modalOpen = false;
        },

        setThemeMode(state, { payload }: PayloadAction<{ mode: ThemeMode | null }>) {
            state.themeMode = payload.mode;
        }
    }
});
