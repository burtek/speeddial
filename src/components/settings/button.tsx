import type { SxProps, Theme } from '@mui/material';
import { IconButton } from '@mui/material';
import { Cog } from 'mdi-material-ui';
import type { FC } from 'react';
import { useCallback } from 'react';

import { actions as configActions } from '@@data/config';
import { useAppDispatch } from '@@data/index';

export const SettingsButton: FC<{ sx?: SxProps<Theme> }> = ({ sx }) => {
    const dispatch = useAppDispatch();
    const onClick = useCallback(() => {
        dispatch(configActions.openModal());
    }, [dispatch]);

    return (
        <IconButton onClick={onClick} sx={sx}>
            <Cog />
        </IconButton>
    );
};
