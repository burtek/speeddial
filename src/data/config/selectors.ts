import { createSelector } from '@reduxjs/toolkit';

import type { RootState } from '..';

import { sliceName } from './slice';


export const getConfig = (state: RootState) => state[sliceName];

export const getThemeMode = createSelector(getConfig, s => s.themeMode);

export const getIsDialogOpen = createSelector(getConfig, s => s.modalOpen);
