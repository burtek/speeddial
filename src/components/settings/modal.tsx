import type { SelectChangeEvent } from '@mui/material';
import { Divider, Dialog, DialogContent, DialogTitle, FormControl, InputLabel, MenuItem, Select } from '@mui/material';
import type { FC } from 'react';
import { useCallback } from 'react';
import { useTranslation } from 'react-i18next';
import { useSelector } from 'react-redux';

import type { ThemeMode } from '@@data/config';
import { actions as configActions } from '@@data/config';
import { getIsDialogOpen, getThemeMode } from '@@data/config/selectors';
import { useAppDispatch } from '@@data/redux-toolkit';


export const SettingsModal: FC = () => {
    const { t } = useTranslation();
    const dispatch = useAppDispatch();

    const dialogOpen = useSelector(getIsDialogOpen);
    const themeMode = useSelector(getThemeMode) ?? 'auto';

    const onChangeThemeMode = useCallback((event: SelectChangeEvent<ThemeMode | 'auto'>) => {
        const value = event.target.value as ThemeMode | 'auto';
        dispatch(configActions.setThemeMode({ mode: value === 'auto' ? null : value }));
    }, [dispatch]);

    const onClose = useCallback(() => {
        dispatch(configActions.closeModal());
    }, [dispatch]);

    return (
        <Dialog
            open={dialogOpen}
            fullWidth
            maxWidth="md"
            onClose={onClose}
        >
            <DialogTitle>{t('configDialog.title')}</DialogTitle>
            <Divider />
            <DialogContent>
                <FormControl fullWidth>
                    <InputLabel id="theme-mode-select-label">{t('configDialog.themeMode.label')}</InputLabel>
                    <Select
                        labelId="theme-mode-select-label"
                        id="theme-mode-select"
                        value={themeMode}
                        label={t('configDialog.themeMode.label')}
                        onChange={onChangeThemeMode}
                    >
                        <MenuItem value="auto">{t('configDialog.themeMode.option.auto')}</MenuItem>
                        <MenuItem value="light">{t('configDialog.themeMode.option.light')}</MenuItem>
                        <MenuItem value="dark">{t('configDialog.themeMode.option.dark')}</MenuItem>
                    </Select>
                </FormControl>
            </DialogContent>
        </Dialog>
    );
};
