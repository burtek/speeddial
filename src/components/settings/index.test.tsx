import userEvent from '@testing-library/user-event';

import { render, fireEvent, waitFor } from 'config/test-utils';

import { SettingsButton } from './button';
import { SettingsModal } from './modal';


function App() {
    return (
        <>
            <SettingsButton />
            <SettingsModal />
        </>
    );
}

describe('settings', () => {
    it('should match snapshot', () => {
        const { queryAllByRole } = render(<App />);

        expect(queryAllByRole('presentation')).toHaveLength(0);

        expect(document.body).toMatchSnapshot();
    });

    describe('dialog management', () => {
        it('should react to click', async () => {
            const { queryByRole, queryAllByRole } = render(<App />);

            fireEvent.click(queryByRole('button') as HTMLButtonElement);

            await waitFor(() => {
                expect(queryAllByRole('presentation').length > 0).toBeTrue();
            });

            expect(document.body).toMatchSnapshot();
        });

        it('should hide dialog', async () => {
            const { queryByRole, queryAllByRole } = render(<App />);

            fireEvent.click(queryByRole('button') as HTMLButtonElement);

            fireEvent.click(document.body.querySelector('.MuiBackdrop-root') as HTMLDivElement);

            await waitFor(() => {
                expect(queryAllByRole('presentation')).toHaveLength(0);
            });

            expect(document.body).toMatchSnapshot();
        });
    });

    describe('settings', () => {
        describe('theme mode', () => {
            it('should change theme mode to light', async () => {
                const user = userEvent.setup();

                const { queryByRole, queryAllByRole } = render(<App />);

                await user.click(queryByRole('button') as HTMLButtonElement);

                await waitFor(() => {
                    expect(queryAllByRole('presentation').length > 0).toBeTrue();
                });

                await user.click(document.body.querySelector('#theme-mode-select') as HTMLDivElement);

                expect(document.body).toMatchSnapshot();
            });
        });
    });
});
