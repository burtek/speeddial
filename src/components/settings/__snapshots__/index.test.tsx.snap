// Vitest Snapshot v1

exports[`settings > dialog management > should hide dialog 1`] = `
<body
  style=""
>
  <div>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CogIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,15.5A3.5,3.5 0 0,1 8.5,12A3.5,3.5 0 0,1 12,8.5A3.5,3.5 0 0,1 15.5,12A3.5,3.5 0 0,1 12,15.5M19.43,12.97C19.47,12.65 19.5,12.33 19.5,12C19.5,11.67 19.47,11.34 19.43,11L21.54,9.37C21.73,9.22 21.78,8.95 21.66,8.73L19.66,5.27C19.54,5.05 19.27,4.96 19.05,5.05L16.56,6.05C16.04,5.66 15.5,5.32 14.87,5.07L14.5,2.42C14.46,2.18 14.25,2 14,2H10C9.75,2 9.54,2.18 9.5,2.42L9.13,5.07C8.5,5.32 7.96,5.66 7.44,6.05L4.95,5.05C4.73,4.96 4.46,5.05 4.34,5.27L2.34,8.73C2.21,8.95 2.27,9.22 2.46,9.37L4.57,11C4.53,11.34 4.5,11.67 4.5,12C4.5,12.33 4.53,12.65 4.57,12.97L2.46,14.63C2.27,14.78 2.21,15.05 2.34,15.27L4.34,18.73C4.46,18.95 4.73,19.03 4.95,18.95L7.44,17.94C7.96,18.34 8.5,18.68 9.13,18.93L9.5,21.58C9.54,21.82 9.75,22 10,22H14C14.25,22 14.46,21.82 14.5,21.58L14.87,18.93C15.5,18.67 16.04,18.34 16.56,17.94L19.05,18.95C19.27,19.03 19.54,18.95 19.66,18.73L21.66,15.27C21.78,15.05 21.73,14.78 21.54,14.63L19.43,12.97Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`settings > dialog management > should react to click 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CogIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,15.5A3.5,3.5 0 0,1 8.5,12A3.5,3.5 0 0,1 12,8.5A3.5,3.5 0 0,1 15.5,12A3.5,3.5 0 0,1 12,15.5M19.43,12.97C19.47,12.65 19.5,12.33 19.5,12C19.5,11.67 19.47,11.34 19.43,11L21.54,9.37C21.73,9.22 21.78,8.95 21.66,8.73L19.66,5.27C19.54,5.05 19.27,4.96 19.05,5.05L16.56,6.05C16.04,5.66 15.5,5.32 14.87,5.07L14.5,2.42C14.46,2.18 14.25,2 14,2H10C9.75,2 9.54,2.18 9.5,2.42L9.13,5.07C8.5,5.32 7.96,5.66 7.44,6.05L4.95,5.05C4.73,4.96 4.46,5.05 4.34,5.27L2.34,8.73C2.21,8.95 2.27,9.22 2.46,9.37L4.57,11C4.53,11.34 4.5,11.67 4.5,12C4.5,12.33 4.53,12.65 4.57,12.97L2.46,14.63C2.27,14.78 2.21,15.05 2.34,15.27L4.34,18.73C4.46,18.95 4.73,19.03 4.95,18.95L7.44,17.94C7.96,18.34 8.5,18.68 9.13,18.93L9.5,21.58C9.54,21.82 9.75,22 10,22H14C14.25,22 14.46,21.82 14.5,21.58L14.87,18.93C15.5,18.67 16.04,18.34 16.56,17.94L19.05,18.95C19.27,19.03 19.54,18.95 19.66,18.73L21.66,15.27C21.78,15.05 21.73,14.78 21.54,14.63L19.43,12.97Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r1:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthMd MuiDialog-paperFullWidth css-rnmm7m-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id=":r1:"
        >
          configDialog.title
        </h2>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="theme-mode-select-label"
            >
              configDialog.themeMode.label
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="theme-mode-select-label theme-mode-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="theme-mode-select"
                role="button"
                tabindex="0"
              >
                configDialog.themeMode.option.auto
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="auto"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    configDialog.themeMode.label
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`settings > settings > theme mode > should change theme mode to light 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CogIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,15.5A3.5,3.5 0 0,1 8.5,12A3.5,3.5 0 0,1 12,8.5A3.5,3.5 0 0,1 15.5,12A3.5,3.5 0 0,1 12,15.5M19.43,12.97C19.47,12.65 19.5,12.33 19.5,12C19.5,11.67 19.47,11.34 19.43,11L21.54,9.37C21.73,9.22 21.78,8.95 21.66,8.73L19.66,5.27C19.54,5.05 19.27,4.96 19.05,5.05L16.56,6.05C16.04,5.66 15.5,5.32 14.87,5.07L14.5,2.42C14.46,2.18 14.25,2 14,2H10C9.75,2 9.54,2.18 9.5,2.42L9.13,5.07C8.5,5.32 7.96,5.66 7.44,6.05L4.95,5.05C4.73,4.96 4.46,5.05 4.34,5.27L2.34,8.73C2.21,8.95 2.27,9.22 2.46,9.37L4.57,11C4.53,11.34 4.5,11.67 4.5,12C4.5,12.33 4.53,12.65 4.57,12.97L2.46,14.63C2.27,14.78 2.21,15.05 2.34,15.27L4.34,18.73C4.46,18.95 4.73,19.03 4.95,18.95L7.44,17.94C7.96,18.34 8.5,18.68 9.13,18.93L9.5,21.58C9.54,21.82 9.75,22 10,22H14C14.25,22 14.46,21.82 14.5,21.58L14.87,18.93C15.5,18.67 16.04,18.34 16.56,17.94L19.05,18.95C19.27,19.03 19.54,18.95 19.66,18.73L21.66,15.27C21.78,15.05 21.73,14.78 21.54,14.63L19.43,12.97Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      >
        <span
          class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
  <div
    aria-hidden="true"
    class="MuiDialog-root MuiModal-root css-zw3mfo-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-yiavyu-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-hz1bth-MuiDialog-container"
      role="presentation"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby=":r3:"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthMd MuiDialog-paperFullWidth css-rnmm7m-MuiPaper-root-MuiDialog-paper"
        role="dialog"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-bdhsul-MuiTypography-root-MuiDialogTitle-root"
          id=":r3:"
        >
          configDialog.title
        </h2>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          class="MuiDialogContent-root css-ypiqx9-MuiDialogContent-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="theme-mode-select-label"
            >
              configDialog.themeMode.label
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="true"
                aria-haspopup="listbox"
                aria-labelledby="theme-mode-select-label theme-mode-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="theme-mode-select"
                role="button"
                tabindex="0"
              >
                configDialog.themeMode.option.auto
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                tabindex="-1"
                value="auto"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined MuiSelect-iconOpen css-bpeome-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    configDialog.themeMode.label
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
  <div
    class="MuiPopover-root MuiMenu-root MuiModal-root css-10nakn3-MuiModal-root-MuiPopover-root-MuiMenu-root"
    id="menu-"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiBackdrop-invisible MuiModal-backdrop css-g3hgs1-MuiBackdrop-root-MuiModal-backdrop"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiPaper-root MuiMenu-paper MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper css-1poimk-MuiPaper-root-MuiMenu-paper-MuiPaper-root-MuiPopover-paper"
      style="opacity: 1; transform: none; min-width: 0; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
      tabindex="-1"
    >
      <ul
        aria-labelledby="theme-mode-select-label"
        class="MuiList-root MuiList-padding MuiMenu-list css-6hp17o-MuiList-root-MuiMenu-list"
        role="listbox"
        tabindex="-1"
      >
        <li
          aria-selected="true"
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters Mui-selected MuiMenuItem-root MuiMenuItem-gutters Mui-selected css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
          data-value="auto"
          role="option"
          tabindex="0"
        >
          configDialog.themeMode.option.auto
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </li>
        <li
          aria-selected="false"
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
          data-value="light"
          role="option"
          tabindex="-1"
        >
          configDialog.themeMode.option.light
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </li>
        <li
          aria-selected="false"
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
          data-value="dark"
          role="option"
          tabindex="-1"
        >
          configDialog.themeMode.option.dark
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </li>
      </ul>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`settings > should match snapshot 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="CogIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,15.5A3.5,3.5 0 0,1 8.5,12A3.5,3.5 0 0,1 12,8.5A3.5,3.5 0 0,1 15.5,12A3.5,3.5 0 0,1 12,15.5M19.43,12.97C19.47,12.65 19.5,12.33 19.5,12C19.5,11.67 19.47,11.34 19.43,11L21.54,9.37C21.73,9.22 21.78,8.95 21.66,8.73L19.66,5.27C19.54,5.05 19.27,4.96 19.05,5.05L16.56,6.05C16.04,5.66 15.5,5.32 14.87,5.07L14.5,2.42C14.46,2.18 14.25,2 14,2H10C9.75,2 9.54,2.18 9.5,2.42L9.13,5.07C8.5,5.32 7.96,5.66 7.44,6.05L4.95,5.05C4.73,4.96 4.46,5.05 4.34,5.27L2.34,8.73C2.21,8.95 2.27,9.22 2.46,9.37L4.57,11C4.53,11.34 4.5,11.67 4.5,12C4.5,12.33 4.53,12.65 4.57,12.97L2.46,14.63C2.27,14.78 2.21,15.05 2.34,15.27L4.34,18.73C4.46,18.95 4.73,19.03 4.95,18.95L7.44,17.94C7.96,18.34 8.5,18.68 9.13,18.93L9.5,21.58C9.54,21.82 9.75,22 10,22H14C14.25,22 14.46,21.82 14.5,21.58L14.87,18.93C15.5,18.67 16.04,18.34 16.56,17.94L19.05,18.95C19.27,19.03 19.54,18.95 19.66,18.73L21.66,15.27C21.78,15.05 21.73,14.78 21.54,14.63L19.43,12.97Z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;
